cdpctl/validation/infra/validate_azure_security_groups.py

```diff
@@ -0,0 +1,434 @@
+# with an authorized and properly licensed third party, you do not
+# have any rights to access nor to use this code.
+#
+# Absent a written agreement with Cloudera, Inc. (“Cloudera”) to the
+# contrary, A) CLOUDERA PROVIDES THIS CODE TO YOU WITHOUT WARRANTIES OF ANY
+# KIND; (B) CLOUDERA DISCLAIMS ANY AND ALL EXPRESS AND IMPLIED
+# WARRANTIES WITH RESPECT TO THIS CODE, INCLUDING BUT NOT LIMITED TO
+# IMPLIED WARRANTIES OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY AND
+# FITNESS FOR A PARTICULAR PURPOSE; (C) CLOUDERA IS NOT LIABLE TO YOU,
+# AND WILL NOT DEFEND, INDEMNIFY, NOR HOLD YOU HARMLESS FOR ANY CLAIMS
+# ARISING FROM OR RELATED TO THE CODE; AND (D)WITH RESPECT TO YOUR EXERCISE
+# OF ANY RIGHTS GRANTED TO YOU FOR THE CODE, CLOUDERA IS NOT LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+# CONSEQUENTIAL DAMAGES INCLUDING, BUT NOT LIMITED TO, DAMAGES
+# RELATED TO LOST REVENUE, LOST PROFITS, LOSS OF INCOME, LOSS OF
+# BUSINESS ADVANTAGE OR UNAVAILABILITY, OR LOSS OR CORRUPTION OF
+# DATA.
+#
+# Source File
+"""Azure Network Security Groups Validations."""
+import ipaddress
+from typing import Any, Dict, List
+
+import pytest
+from azure.core.exceptions import ResourceNotFoundError
+from azure.mgmt.network import NetworkManagementClient
+
+from cdpctl.validation import fail, get_config_value, validator, warn
+from cdpctl.validation.azure_utils import get_client
+from cdpctl.validation.infra.issues import (
+    AZURE_CDP_CIDR_ACCESS_NOT_ALLOWED_FOR_PORT,
+    AZURE_NSG_NOT_FOUND,
+    AZURE_SUBNETS_NOT_COVERED_BY_NSG,
+    AZURE_SUBNETS_NOT_VALID_NSG,
+    AZURE_VNET_NOT_FOUND,
+)
+
+_info = {}
+
+
+@pytest.fixture(autouse=True, name="azure_network_client")
+def azure_network_client_fixture(config: Dict[str, Any]) -> NetworkManagementClient:
+    """Return an Azure Client."""
+    return get_client("network", config)
+
+
+def _load_vnet_info(vnet_name, resource_group_name, azure_network_client):
+    if "vnet" not in _info:
+        try:
+            _info["vnet"] = azure_network_client.virtual_networks.get(
+                virtual_network_name=vnet_name, resource_group_name=resource_group_name
+            )
+        except ResourceNotFoundError:
+            fail(AZURE_VNET_NOT_FOUND, [vnet_name, resource_group_name])
+
+
+@pytest.fixture(name="azure_resource_group_name")
+def azure_resource_group_name_fixture(config: Dict[str, Any]):
+    """Return the resource group name."""
+    return get_config_value(config=config, key="infra:azure:metagroup:name")
+
+
+@pytest.fixture(name="azure_vnet_name")
+def azure_vnet_name_fixture(config: Dict[str, Any]):
+    """Return the vnet name."""
+    return get_config_value(config=config, key="infra:vpc:name")
+
+
+@pytest.fixture(name="azure_vnet_info")
+def azure_vnet_info_fixture(config, azure_network_client):
+    """Return the vnet info."""
+    vnet_name = get_config_value(config=config, key="infra:vpc:name")
```

Comment:
curious, can a fixture not have another fixture as an argument? (azure_vnet_name)

Reply:
No it can. I just missed it.